### PR TITLE
[GLUTEN-7741][VL] Fix deprecated actions/upload-artifact version issue when building bundle package

### PR DESCRIPTION
**Pull request by wangyum**
## What changes were proposed in this pull request?

https://github.com/wangyum/incubator-gluten/actions/runs/11949609833:
```
This request has been automatically failed because it uses a deprecated version of `actions/upload-artifact: v2`. Learn more: https://github.blog/changelog/2024-02-13-deprecation-notice-v1-and-v2-of-the-artifact-actions/
```

## How was this patch tested?

manual tests: https://github.com/wangyum/incubator-gluten/actions/runs/11965336260/job/33359296015

### CODE REVIEW

File: .github/workflows/build_bundle_package.yml [hunk 1]
```diff
@@ -54,16 +54,19 @@ jobs:
           cd $GITHUB_WORKSPACE/
           bash dev/ci-velox-buildstatic-centos-7.sh
       - name: Upload native libs
-        uses: actions/upload-artifact@v2
+        uses: actions/upload-artifact@v4
```
Comment (zhouyuan) [inline .github/workflows/build_bundle_package.yml:57]: It seems V3 should work on Centos7
https://github.com/apache/incubator-gluten/blob/main/.github/workflows/velox_backend.yml#L85

File: .github/workflows/build_bundle_package.yml [hunk 2]
```diff
@@ -94,9 +97,10 @@ jobs:
           export PATH=${PATH}:${MAVEN_HOME}/bin && \
           mvn clean install -P${{ github.event.inputs.spark }} -Dhadoop.version=${{ github.event.inputs.hadoop }} -Pbackends-velox -Pceleborn -Puniffle -DskipTests -Dmaven.source.skip
       - name: Upload bundle package
-        uses: actions/upload-artifact@v2
+        uses: actions/upload-artifact@v3
```
Comment (zhouyuan) [inline .github/workflows/build_bundle_package.yml:100]: may also need to fix the `download-artifact`

Comment (wangyum) [reply .github/workflows/build_bundle_package.yml:99]: yes

File: .github/workflows/build_bundle_package.yml [hunk 3]
```diff
@@ -72,17 +72,19 @@ jobs:
...
       - name: Download All Arrow Jar Artifacts
-        uses: actions/download-artifact@v2
+        uses: actions/download-artifact@v3
         with:
           name: velox-arrow-jar-centos-7-${{github.sha}}
           path: /root/.m2/repository/org/apache/arrow/
       - name: Setup java and maven
         run: |
+          sed -i 's/mirrorlist/#mirrorlist/g' /etc/yum.repos.d/CentOS-* && \
+          sed -i 's|#baseurl=http://mirror.centos.org|baseurl=http://vault.centos.org|g' /etc/yum.repos.d/CentOS-* && \
```
Comment (wangyum) [inline .github/workflows/build_bundle_package.yml:87]: These changes also needed:
image

**Review by zhouyuan — APPROVED** [review]
👍

**Pull request merged**